src/Microsoft.IdentityModel.Clients.ActiveDirectory/Core/Exceptions/CoreErrorMessages.cs

```diff
@@ -115,5 +117,14 @@ internal class CoreErrorMessages
 
         public const string NoRedirectUri = "No redirectUri was configured. ADAL does not provide any defaults.";
 
+        public const string CustomWebUiReturnedInvalidUri = "ICustomWebUi returned an invalid uri - it is empty or has no query.";
+
+        public static string CustomWebUiRedirectUriMismatch(string expectedUri, string actualUri)
```

Comment:
Since you're in here, can you fix the other error messages above (like MissingEntitlements and DefaultRedirectUriIsINvalid) to be functions like this one instead of formatting them out of band?  thanks!

Reply:
oh, this is ADAL.  you can still do it if you want, but not as important since there's a lot of other legacy stuff related to this.

---
In reply to: [262185082](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1511#discussion_r262185082) [](ancestors = 262185082)